annotation-utils/src/main/java/edu/isi/vista/annotationutils/GetAnnotationDurations.kt

```diff
@@ -0,0 +1,217 @@
+    var patternMatch: MatchResult? = null
+    val projectNamePattern = if (projectName.startsWith("ACE")) {
+        // ACE project name format: ACE-Hyphenated.Event-Type-user_name
+        Regex(pattern = """(ACE-[a-zA-Z]*\.[a-zA-Z]*-?[a-zA-Z]*?)-(.*)""")
+    } else if (projectName.startsWith("CORD19")) {
+        // CORD-19 project name format:
+        // CORD19-RelationType-user_name
+        Regex(pattern = """(CORD19-[a-zA-Z]*)-(.*)""")
+    } else {
+        // gigaword project name format:
+        // optionallanguage-Standard.EventType-user_name
+        Regex(pattern = """([a-z]*?-?.*)-(.*)""")
+    }
+    if (projectNamePattern.containsMatchIn(projectName)) {
+        patternMatch = projectNamePattern.find(projectName)
+    }
+    if (patternMatch != null) {
+        eventType = patternMatch.groups[1]!!.value
+        username = patternMatch.groups[2]!!.value
+    }
+    return if (username != null && eventType != null) {
+        ProjectInfo(username, eventType, 0)
+    } else {
+        null
+    }
+}
+
+private fun convertEventsToJson(log: File): List<JsonNode> {
+    // Each line in an event.log represents an action in
+    // Inception, and these can be converted
+    // to JSON objects.
+    val eventObjectMapper = ObjectMapper()
+    val logEvents = log.readLines()
+    return logEvents.map {eventObjectMapper.readTree(it)}
+}
+
+private fun getTimeOnProject(logEvents: List<JsonNode>, username: String): Long {
+    // Get times (in seconds) spent in each document by running
+    // through each Inception event recorded in event.log
+    var currentDocument: String? = null  // some events have no document field
+    var previousTime: Long = 0
+    var documentTimeElapsed: Long = 0
+    val documentTimeMap = mutableMapOf<String, Long>().withDefault { 0 }
+    for (event in logEvents) {
+        val documentName = event.get("document_name")?.toString()?.removeSurrounding("\"")
+        val user = event.get("user")?.toString()?.removeSurrounding("\"")
+        // Only deal with events that have a "document_name" field
+        // and were completed by the user
+        // (admin monitoring activity also gets recorded)
+        if (documentName != null && user == username) {
+            // Timestamps are in Unix time (milliseconds)
+            val timestamp = event.get("created").toString().toLong()
+            val timeSinceLastEvent = timestamp - previousTime
+            if (previousTime == 0.toLong()) {
+                // This is the first event in the log
+                currentDocument = documentName
+            }
+            // If there is a relatively long gap between events,
+            // it may suggest that the annotator took a break.
+            // Check if the time elapsed since the last event
+            // is less than 5 minutes (300,000 milliseconds) -
+            // else, don't add it to the overall time spent
+            // on the document.
+            if (timeSinceLastEvent < 300000) {
```

Comment:
Does five minutes seem like an appropriate threshold? I want to account for breaks, but I also want to check for slow annotators. Considering this would be the time between individual events (marking a span, drawing relations, etc.) maybe this should be lower.

Reply:
I would lower it to 2 minutes.